Update YAMLSyntax.rst with yamllint (#66622)
ansible/ansible: docs/docsite/rst/reference_appendices/YAMLSyntax.rst

    knows_oop: True
    likes_emacs: TRUE
    uses_cvs: false
    

> If you care about code-quality by using `yamllint`, or `molecule`, then you will use the lowercase `true` and `false`.

Reviewer: Recommend re-phrasing this to something like "Use lowercase 'true' or 'false' for boolean values in dictionaries to ensure good code quality."

Author: Amended.

Reviewer: Looks good, thanks @bbaassssiiee

    likes_emacs: TRUE
    uses_cvs: false

> Use lowercase 'true' or 'false' for boolean values in dictionaries to ensure good code quality, testable by using yamllint.

Reviewer 2: totally disagree with that statement, keeping a consistent boolean value is good, the choosing lower case specifically is not really inherently better then most other choices

Author: "Ansible-lint, Yamllint and Flake8 kill opinion wars". (Luke Murphy, author of molecule).

Author: Just try yamllinting with the default options.

Reviewer 2: a) does not kill opinion wars, just transfers them to the linter options
b) i already know what yamllint does and it is one of the options we actually override

Author: Is that documented?

Author: We skip some rules because we don't like them.

Reviewer 2: most linters allow you to skip rules because in many case it is just a convention decided by the linting tool author and does not have solid reasons to go one way or another. 

As a general rule we try not to impose opinions to playbook authors, which is why i have a problem with this sentence.

Author: More matter of fact now:
> Use lowercase 'true' or 'false' for boolean values in dictionaries to ensure a code quality testable with default yamllint options.

> Use lowercase 'true' or 'false' for boolean values in dictionaries to ensure a code quality testable with default yamllint options.

Reviewer 3: ```suggestion
Use lowercase 'true' or 'false' for boolean values in dictionaries if you want to be compatible with default yamllint options.
```

Reviewer 3: making this ^^^ as neutral as possible :-)
